jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -899,11 +899,15 @@ def _update_stretch_curve(self, msg=None):
         # show "knot" locations if the stretch_function is a spline
         if isinstance(stretch, SplineStretch) and self.stretch_curve_visible:
             knot_mark = self.stretch_histogram.marks['stretch_knots']
+            # ensure knot x values are within 0-1 range after scaling
+            limited_knots_x = np.clip(np.asarray(stretch._x), 0, 1)

```

Comment:
this seems to mostly work, but dragging into the colorbar still has unexpected behavior.  Can this be enforced at the tool level instead/as well?